### PR TITLE
Xample: add svc-xample-j integration test

### PR DESCRIPTION
**Pull request by yoomlam**
## What was the problem?

Developer wants an integration test for `svc-xample-j` microservice to continue #1594 

Associated tickets or Slack threads:
- #1763

## How does this fix it?

Add JUnit integration test to `svc-xample-j` microservice

## How to test this PR
```sh
# Start RabbitMQ:
docker compose up -d rabbitmq-service

# Run the tests:
./gradlew :domain-xample:svc-xample-j:integrationTest
```

### CODE REVIEW

File: domain-xample/svc-xample-j/src/integrationTest/java/gov/va/vro/services/xample/XampleJavaMicroserviceTest.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+ * <p>
+ * Run these tests:
+ * ./gradlew :domain-xample:svc-xample-j:integrationTest
+ */
+@SpringBootTest
+@Slf4j
+public class XampleJavaMicroserviceTest {
+
+  @Autowired
+  private RabbitTemplate rabbitTemplate;
```
Comment (yoomlam) [inline domain-xample/svc-xample-j/src/integrationTest/java/gov/va/vro/services/xample/XampleJavaMicroserviceTest.java:35]: Don't need to create separate MQ producer and consumer if using Spring's `RabbitTemplate`.

**Comment by yoomlam** [comment]
[CI: Xample domain integration test](https://github.com/department-of-veterans-affairs/abd-vro/actions/runs/5413256858) works

File: domain-xample/svc-xample-j/src/integrationTest/java/gov/va/vro/services/xample/XampleJavaMicroserviceTest.java (hunk 1 above)
Comment (dfitchett) [reply domain-xample/svc-xample-j/src/integrationTest/java/gov/va/vro/services/xample/XampleJavaMicroserviceTest.java:35]: 🤯

File: domain-xample/svc-xample-j/src/integrationTest/java/gov/va/vro/services/xample/XampleJavaMicroserviceTest.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  void setUp() {
+    rabbitAdmin.purgeQueue(queueName, true);
+  }
+
+  @AfterEach
+  void tearDown() {
+    rabbitAdmin.purgeQueue(queueName, true);
+  }
+
+  SomeDtoModel request = SomeDtoModel.builder().resourceId("320").diagnosticCode("B").build();
```
Comment (dfitchett) [inline domain-xample/svc-xample-j/src/integrationTest/java/gov/va/vro/services/xample/XampleJavaMicroserviceTest.java:50]: private final 👍

File: domain-xample/svc-xample-j/src/integrationTest/java/gov/va/vro/services/xample/XampleJavaMicroserviceTest.java [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    assertEquals(StatusValue.DONE.toString(), response.getStatus());
+    assertEquals(200, response.getHeader().getStatusCode());
+    assertNull(response.getHeader().getStatusMessage());
+  }
+
+  @Autowired
+  private ObjectMapper objectMapper;
+
+  @Test
+  void sendMessageToGetJson() throws IOException {
```
Comment (engineer-plus-plus) [inline domain-xample/svc-xample-j/src/integrationTest/java/gov/va/vro/services/xample/XampleJavaMicroserviceTest.java:67]: This wasn't requested in the story.  More importantly, it is not clear what it's doing (to me).  I recommend to either make it more clear what use case the test asserts; or merge it with happy use case test; or remove it.

**Review by engineer-plus-plus — CHANGES_REQUESTED** [review]
See in-line comment please

File: domain-xample/svc-xample-j/src/integrationTest/java/gov/va/vro/services/xample/XampleJavaMicroserviceTest.java (hunk 3 above)
Comment (yoomlam) [reply domain-xample/svc-xample-j/src/integrationTest/java/gov/va/vro/services/xample/XampleJavaMicroserviceTest.java]: Updated code and added comments for clarity.

File: domain-xample/svc-xample-j/src/integrationTest/java/gov/va/vro/services/xample/XampleJavaMicroserviceTest.java [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    assertEquals(request.getDiagnosticCode(), response.getDiagnosticCode());
+    assertEquals(StatusValue.DONE.toString(), response.getStatus());
+    assertEquals(200, response.getHeader().getStatusCode());
+    assertNull(response.getHeader().getStatusMessage());
+  }
+
+  @Autowired
+  private ObjectMapper objectMapper;
+
+  // Can also send a plain JSON String without any RabbitTemplate auto-conversion of the msg body
```
Comment (engineer-plus-plus) [inline domain-xample/svc-xample-j/src/integrationTest/java/gov/va/vro/services/xample/XampleJavaMicroserviceTest.java:66]: 👍

**Review by engineer-plus-plus — APPROVED** [review]

**Pull request merged**